sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -443,13 +445,22 @@ private Read() {}
     }
 
     /**
+     * Returns a {@link PTransform} that writes Avro file(s) using specified codec.
+     */
+    public static Bound<GenericRecord> withCodec(CodecFactory codec) {
+      return new Bound<>(GenericRecord.class).withCodec(codec);
+    }
+
+    /**
      * A {@link PTransform} that writes a bounded {@link PCollection} to an Avro file (or
      * multiple Avro files matching a sharding pattern).
      *
      * @param <T> the type of each of the elements of the input PCollection
      */
     public static class Bound<T> extends PTransform<PCollection<T>, PDone> {
       private static final String DEFAULT_SHARD_TEMPLATE = ShardNameTemplate.INDEX_OF_MAX;
+      private static final SerializableAvroCodecFactory DEFAULT_CODEC =
+          new SerializableAvroCodecFactory(CodecFactory.deflateCodec(6));
```

Comment:
Is this the current default, or a behavior change? (I assume the actual current default is that we use the `null` codec?) If so, put this behavior change in the commit message :)